actors/builtin/reward/reward_logic.go

```diff
@@ -0,0 +1,72 @@
+package reward
+
+import (
+	abi "github.com/filecoin-project/specs-actors/actors/abi"
+	big "github.com/filecoin-project/specs-actors/actors/abi/big"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+)
+
+var BaselinePowerAt = func(epoch abi.ChainEpoch) abi.StoragePower {
+	return big.NewInt(1 << 40)
+}
+
+// Computes RewardTheta, result is in Q.128 format
+func computeRTheta(effectiveNetworkTime abi.ChainEpoch, cumsumRealized, cumsumBaseline big.Int) big.Int {
+	var rewardTheta big.Int
+	if effectiveNetworkTime != 0 {
+		rewardTheta = big.NewInt(int64(effectiveNetworkTime)) // Q.0
+		rewardTheta = big.Lsh(rewardTheta, precision)         // Q.0 => Q.128
+		diff := big.Sub(cumsumBaseline, cumsumRealized)
+		diff = big.Lsh(diff, precision)                             // Q.0 => Q.128
+		diff = big.Div(diff, BaselinePowerAt(effectiveNetworkTime)) // Q.128 / Q.0 => Q.128
+		rewardTheta = big.Sub(rewardTheta, diff)                    // Q.128
+	} else {
+		// sepecial case for initialization
+		rewardTheta = big.Zero()
+	}
+	return rewardTheta
+}
+
+// These numbers are placeholders, but should be in units of attoFIL, 10^-18 FIL
+var SimpleTotal = big.Mul(big.NewInt(100e6), big.NewInt(1e18))   // 100M for testnet, PARAM_FINISH
+var BaselineTotal = big.Mul(big.NewInt(900e6), big.NewInt(1e18)) // 900M for testnet, PARAM_FINISH
+
+var (
+	// parameters in Q.128 format
+	// lambda = tropicalYearInSeconds/blockDelay*ln(2)
+	lambda, _ = big.FromString("186857370934482378542986172834581")
```

Comment:
If possible, I would really like to compute this in code. If that's not reasonable, please add a big comment next to `builtin.EpochDurationSeconds` that points to this number needing to be recomputed if the delay changes.

Reply:
Not really possible to compute it in code (due to `ln(2)`), nor practical.
I will give exact formula.